### DISCUSSION TITLE
Using GMM in configuration.yml gives NSE estimation but cannot find the simulated streamflow as in linear.yml

### DISCUSSION THREAD

**Discussion by maby200** [neuralhydrology/neuralhydrology#189] [Q&A] [answered]
Hello ,

I have been using the GMM inside my `configuration.yml` for uncertainty estimation. The results are ok. But I found there are NSE values for the current set of simulation.

If there would be for the regular linear configuration, like in [Kratzert, 2018](https://hess.copernicus.org/articles/22/6005/2018/hess-22-6005-2018.pdf) there is a single array for output, and as consequence, the NSE based in this simulated and observed makes sense.

But in the results of GMM, CMAL or UMAL, there are various sets of simulated arrays, which is normal based on their nature.

My question is, how the NSE was calculated? where is that single array for the simulated discharge, if exists?

Hope you can help me.

Thanks in advance :)

**Comment by gauchm** [comment, marked as answer]
The metrics are calculated on the mean of all samples, see here: https://github.com/neuralhydrology/neuralhydrology/blob/master/neuralhydrology/evaluation/tester.py#L307

So you can easily get that single simulation timeseries by applying the mean over the "samples" dimension.

**Reply by maby200** [reply]
Thank you very much!
